### PR TITLE
Add: Draw Plot border when holding Sprayonator

### PR DESCRIPTION
**Pull request by superhize**
I moved plot border render logic into GardenPlotAPI to avoid boilerplate

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/SprayFeatures.kt [hunk 1]
```diff
@@ -57,5 +63,13 @@ class SprayFeatures {
         config.position.renderString(display, posLabel = "Pest Spray Selector")
     }
 
+    @SubscribeEvent
+    fun onWorldRender(event: LorenzRenderWorldEvent) {
+        if (!config.drawPlotsBorderWhenInHands) return
+        if (InventoryUtils.getItemInHand()?.getInternalName()?.asString() != "SPRAYONATOR") return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/pests/SprayFeatures.kt]: You can use `InventoryUtils.itemInHandId` here.

File: src/main/java/at/hannibal2/skyhanni/features/garden/GardenAPI.kt [hunk 2]
```diff
@@ -230,6 +236,88 @@ object GardenAPI {
         return tier
     }
 
+    fun LorenzRenderWorldEvent.renderPlot(plot: GardenPlotAPI.Plot, lineColor: Color, cornerColor: Color) {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/GardenAPI.kt]: I'd say putting it into `GardenPlotAPI` feels better.

File: src/main/java/at/hannibal2/skyhanni/features/garden/GardenAPI.kt [hunk 3]
```diff
@@ -70,6 +75,7 @@ object GardenAPI {
         "ROOKIE_HOE",
         "BINGHOE"
     )
+    private val LINE_COLOR = LorenzColor.YELLOW.toColor()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/GardenAPI.kt]: Do we need this?

**Review by hannibal002 — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/SprayFeatures.kt (hunk 1 above)
Comment (superhize) [reply src/main/java/at/hannibal2/skyhanni/features/garden/pests/SprayFeatures.kt]: I forgot, my bad

File: src/main/java/at/hannibal2/skyhanni/features/garden/GardenAPI.kt (hunk 3 above)
Comment (superhize) [reply src/main/java/at/hannibal2/skyhanni/features/garden/GardenAPI.kt]: We don't, forgot to remove this line

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**